src/java.base/share/native/launcher/main.c

```diff
@@ -167,6 +167,49 @@ static void setup_sighandler() {
     }
 }
 
+static const char *last_pid_filename = "/proc/sys/kernel/ns_last_pid";
+
+static int get_last_pid() {
+    FILE *last_pid_file = fopen(last_pid_filename, "r");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
+        return -1;
+    }
+    int last_pid;
+    if (0 >= fscanf(last_pid_file, "%d", &last_pid)) {
+        fclose(last_pid_file);
+        perror("last_pid_file fscanf");
+        last_pid = -1;
+    }
+    fclose(last_pid_file);
+    return last_pid;
+}
+
+static void set_last_pid(int pid) {
+    FILE *last_pid_file = fopen(last_pid_filename, "w");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
+        return;
+    }
+    if (0 > fprintf(last_pid_file, "%d", pid)) {
+        perror("last_pid_file fprintf");
+    }
+    fclose(last_pid_file);
+}
+
+static void spin_last_pid(int pid) {
+    for (pid_t child = fork(); child < (pid_t)pid; child = fork()) {
```

Comment:
Although this part is not peformance critical, in my previous experiments, fork() was considerably more expensive compared to some minimalistic thread or vfork(). The best I found was `syscall(SYS_clone, SIGCHLD, NULL, NULL, 0)`. Could you check can that be made not too ugly? Maybe pthread_create?

Reply:
ok, I will check